test/get_authors_test.rb

```diff
@@ -53,6 +48,17 @@ def test_markdown_center
     assert_match %Q|{:center: style="text-align: center"}\n#{text}\n{:center}|, Markdown.center(text)
   end
 
+  def test_git_log
+    method_output = Git.log File.join(@subdir, 'README.md')
+    assert_equal "Your Name;you@example.com", "#{method_output}"
```

Comment:
Isn't the notation `"#{...}"` redundant?

Reply:
It is not, the output is not transformed into string if it is not done like that. If it would be absent then tests output what we are not testing. Just FYI minitest output if it would be just bare "method_output":
~~~
ruby test/get_authors_test.rb 
Run options: --seed 8429

# Running:

S..SF

Failure:
TestGetAuthors#test_git_log [test/get_authors_test.rb:53]:
--- expected
+++ actual
@@ -1 +1 @@
-"Your Name;you@example.com"
+#<Git::Log:0xXXXXXX @log="Your Name;you@example.com">



bin/rails test test/get_authors_test.rb:51

.S.

Finished in 0.222792s, 35.9080 runs/s, 35.9080 assertions/s.
8 runs, 8 assertions, 1 failures, 0 errors, 3 skips
~~~

Reply:
What about `method_output.to_s` ?

Reply:
Unfortunately that doesn't work either. Minitest output:
~~~
$ ruby test/get_authors_test.rb 
Run options: --seed 5147

# Running:

.F

Failure:
TestGetAuthors#test_git_log [test/get_authors_test.rb:53]:
--- expected
+++ actual
@@ -1 +1,2 @@
-"Your Name;you@example.com"
+# encoding: ASCII-8BIT
+"#<Git::Log:0xXXXXXX>"



bin/rails test test/get_authors_test.rb:51

..SS.S

Finished in 0.249783s, 32.0278 runs/s, 32.0278 assertions/s.
8 runs, 8 assertions, 1 failures, 0 errors, 3 skips
~~~

Reply:
Well, since it's just a test suite, I do not mind. But it's a quite strange behavior from an object.